src/header/mod.rs

```diff
@@ -546,6 +551,32 @@ impl fmt::Debug for Headers {
     }
 }
 
+#[cfg(feature = "compat")]
+impl From<http_types::HeaderMap> for Headers {
+    fn from(header_map: http_types::HeaderMap) -> Headers {
+        let mut headers = Headers::new();
+        for (name, value) in header_map.iter() {
+            headers.set_raw(name.as_str().to_string(), value.as_bytes());        
+        }
+        headers
+    }
+}
+
+#[cfg(feature = "compat")]
+impl From<Headers> for http_types::HeaderMap {
+    fn from(headers: Headers) -> http_types::HeaderMap {
+        let mut header_map = http_types::HeaderMap::new();
+        for header in headers.iter() {
+            let name = http_types::header::HeaderName::from_bytes(header.name().as_bytes())
+                .expect("attempted to convert invalid header name");
+            let value = http_types::header::HeaderValue::from_bytes(header.value_string().as_bytes())
```

Comment:
This should use `header.raw()` instead of `value_string()`. The `value_string` method can panic, even though the raw bytes were a valid header value.

Additionally, when using `raw`, that should mean a 2nd loop. The `name` can be constructed once, and used for each call to `append` for each item `raw`.

Reply:
Makes sense.